Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/AssetFolderThumbnailView.cpp

```diff
@@ -790,9 +790,10 @@ namespace AzQtComponents
         auto idx = indexAtPos(p);
         if (idx.isValid())
         {
+            bool isRightClick = event->button() == Qt::RightButton;
             if (selectionMode() == ExtendedSelection && QGuiApplication::queryKeyboardModifiers() == Qt::ControlModifier)
             {
-                selectionModel()->select(idx, QItemSelectionModel::SelectionFlag::Toggle);
+                (isRightClick ? selectionModel()->select(idx, QItemSelectionModel::SelectionFlag::Select) : selectionModel()->select(idx, QItemSelectionModel::SelectionFlag::Toggle));
```

Comment:
nit: Would be good to do this instead to improve readability a bit.

```cpp
auto selectionFlag = isRightClick ? QItemSelectionModel::SelectionFlag::Select : QItemSelectionModel::SelectionFlag::Toggle;
selectionModel()->select(idx, selectionFlag);
```

Reply:
I agree. While this should work, you should either explicitly use an if statement and curly braces for the function calls or move the condition into the second parameter for the selection flag. The above recommendation is the cleanest.

Reply:
Done :)